File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    widths = []
    heights = []
    
    for size in sizes:
        if size[0] < size[1]:
            size[0], size[1] = size[1], size[0]
        widths.append(size[0])
        heights.append(size[1])
    
    return max(widths)*max(heights)

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int max_w = 0, max_h = 0;
        
        for(int[] size : sizes) {
            int width = Math.max(size[0], size[1]);
            int height = Math.min(size[0], size[1]);
            max_w = Math.max(width, max_w);
            max_h = Math.max(height, max_h);
        }
        
        return (max_w * max_h);
    }
}